File: src/main/java/net/onelitefeather/pandorascluster/listener/PlayerConnectionListener.java
package net.onelitefeather.pandorascluster.listener;

import net.onelitefeather.pandorascluster.api.PandorasClusterApi;
import net.onelitefeather.pandorascluster.land.player.LandPlayer;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;
import org.bukkit.event.player.PlayerQuitEvent;

import javax.imageio.plugins.tiff.TIFFTagSet;

public record PlayerConnectionListener(PandorasClusterApi api) implements Listener {

    @EventHandler
    public void handlePlayerJoin(PlayerJoinEvent event) {
        Player player = event.getPlayer();
        this.api.registerPlayer(player.getUniqueId(), player.getName(), success -> {
            if(success) {
                player.sendMessage("Your playerdata was successfully created!");
            } else {
                player.sendMessage("Your playerdata could not be created!");
            }
        });
    }

    @EventHandler
    public void onPlayerQuit(PlayerQuitEvent event) {
        LandPlayer landPlayer = this.api.getLandPlayerService().getLandPlayer(event.getPlayer().getUniqueId());
        if (landPlayer != null) {
            this.api.getLandPlayerService().updateLandPlayer(landPlayer);
        }
    }
}

File: src/main/kotlin/net/onelitefeather/pandorascluster/listener/PlayerConnectionListener.kt
package net.onelitefeather.pandorascluster.listener

import net.onelitefeather.pandorascluster.api.PandorasClusterApi
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent
import org.bukkit.event.player.PlayerQuitEvent

class PlayerConnectionListener(private val api: PandorasClusterApi) : Listener {

    @EventHandler
    fun handlePlayerJoin(event: PlayerJoinEvent) {
        val player = event.player
        this.api.registerPlayer(player.uniqueId, player.name) { success: Boolean ->
            if (success) {
                player.sendMessage("Your playerdata was successfully created!")
            }
        }
    }

    @EventHandler
    fun onPlayerQuit(event: PlayerQuitEvent) {
        val landPlayer = api.getLandPlayer(event.player.uniqueId) ?: return
        api.getLandPlayerService().updateLandPlayer(landPlayer)
    }
}